Helpers/CommonMenuItemPartMigrator.cs

```diff
@@ -0,0 +1,128 @@
+            _actionContextAccessor = actionContextAccessor;
+            _httpContextAccessor = httpContextAccessor;
+            _logger = logger;
+        }
+
+        public async Task GetCommonMenuitemDataAsync()
+        {
+            var contentItems = await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.Latest && x.ContentType == "Menu")
+                .ListAsync();
+
+            foreach (var contentItem in contentItems)
+            {
+                var menuItemsListPart = contentItem.As<MenuItemsListPart>();
+
+                foreach (var linkMenuItem in menuItemsListPart.MenuItems)
+                {
+                    var linkDestinationPart = new LinkDestinationPart();
+
+                    try
+                    {
+                        if (linkMenuItem.As<CommonMenuItem>().ExternalUrl.Text != null)
+                        {
+                            linkDestinationPart = new LinkDestinationPart { DestinationUrl = linkMenuItem.As<CommonMenuItem>().ExternalUrl }; ;
+                        }
+                        else
+                        {
+                            var host = _httpContextAccessor.HttpContext.Request.Host.Value;
+                            var path = _httpContextAccessor.HttpContext.Request.Path.Value != null ? _httpContextAccessor.HttpContext.Request.Path.Value.Replace("admin","") : "/" ;
+
+                            linkDestinationPart = new LinkDestinationPart
+                            {
+                                DestinationUrl = new TextField { Text = $"//{host}{path}{await ContentItemIdToUrl(linkMenuItem)}" }
+                            };
+                        }
+
+                        var linkBehaviourPart = new LinkBehaviourPart
+                        {
+                            ClickEvent = linkMenuItem.As<CommonMenuItem>().OnClick,
+                            OpenIn = new TextField { Text = linkMenuItem.As<CommonMenuItem>().OpenNewTab.Value ? "_blank" : "_self" }
+                        };
+
+                        linkMenuItem.Apply(nameof(LinkDestinationPart), linkDestinationPart);
+                        linkMenuItem.Apply(nameof(LinkBehaviourPart), linkBehaviourPart);
+                        ContentExtensions.Apply(linkMenuItem, linkMenuItem);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, $"Failed to update menu item: {linkMenuItem.ContentType} ID : {linkMenuItem.ContentItemId}");
+                        continue;
+                    }
+                }
+
+                contentItem.Apply(nameof(MenuItemsListPart), menuItemsListPart);
+                await _contentManager.UpdateAsync(contentItem);
+                await _contentManager.PublishAsync(contentItem);
+            };
+
+        }
+
+        public async Task<string> ContentItemIdToUrl(ContentItem linkMenuItem)

```

Comment:
Change method argument to a string that represents the `contentItemId`. There is no point passing in the whole content item because you're only using the content item ID. Makes this method much cleaner.

Reply:
Method is async, which means method name should be postfixed with "Async".

Reply:
Method can be `private` too.